src/components/dagshub/data-engine/singleFileViewModal/SingleFileViewModal.tsx

```diff
@@ -0,0 +1,284 @@
+                                            display: 'flex',
+                                            width: '100%',
+                                            height: '52px',
+                                            padding: '8px',
+                                            justifyContent: 'center',
+                                            alignItems: 'center',
+                                            boxSizing: "border-box"
+                                        }}
+                                    >
+                                        <Box
+                                            sx={{
+                                                display: 'flex',
+                                                justifyContent: 'center',
+                                                alignItems: 'center',
+                                                gap: '8px',
+                                            }}
+                                        >
+                                            <Button
+                                                onClick={() => {
+                                                    setActiveSingleItem(items[activeSingleItemIndex - 1]?.node);
+                                                    setActiveSingleItemIndex(activeSingleItemIndex - 1);
+                                                }}
+                                                label={''}
+                                                iconRight={
+                                                    <Icon
+                                                        icon={'outline-arrow-sm-left'}
+                                                        width={11.67}
+                                                        height={10}
+                                                        fill={'#172D32'}
+                                                    />
+                                                }
+                                                variant={ButtonVariant.Secondary}
+                                                disabled={activeSingleItemIndex == 0}
+                                            />
+                                            <Button
+                                                onClick={() => {
+                                                    //if im close to the last fetched item, fetch more, so it won't get stack
+                                                    if (activeSingleItemIndex + 10 == items?.length && hasMoreItemsToFetch) {
+                                                        loadMoreItems();
+                                                    }
+                                                    setActiveSingleItem(items[activeSingleItemIndex + 1]?.node);
+                                                    setActiveSingleItemIndex(activeSingleItemIndex + 1);
+                                                }}
```

Comment:
This entire function should be passed from outside, as `getNextItem`. That function should return the next `itemData` and there should be an `itemData` state that is initialized with some initial `itemData`. 

There is no reason this component should know anything about the other items.

Note:
`itemData` should also contain `hasNext, hasPrevious` to know if the next / prev button should be disabled or not.